### DISCUSSION TITLE
Feature Request: option to pull reservation from end of CIDR block

### DISCUSSION THREAD

**Discussion by whizzer929** [Azure/ipam#72] [Ideas]
Hi DCMattyG, This would be a nice to have.  Have an optional param in the reservation API to pull the requested subnet from the end of the CIDR block, instead of the beginning.  This would allow us to fill in blocks from opposite ends of the IP space.

**Comment by DCMattyG** [comment]
I'm curious on this one @whizzer929...generally with subnetting like this you want to keep the assignments close to the front of the block to leave larger space at the end, in case the need arises for some sizeable CIDR requests.

Can you help me understand what the logic you would use to decide whether or not to request a CIDR from the beginning or end of the block?

**Reply by whizzer929** [reply]
It's because we can't estimate our workloads very accurately and we didn't want to paint ourselves into a corner by preassigning large IP blocks.  So we use a very large block and assign workloads from opposite ends and where the IP space gets divided can be moved toward one workload or the other.  Adjusting rules, etc as needed.  Yeah, it's a small ask and now that the IPAM API can use multiple blocks, I think we can get around this by creating more IPAM blocks per workload and use our same model.  And just add more IPAM blocks when needed.  I'll leave it to you as to whether or not this might be worthwhile.  Thanks again.

**Reply by DCMattyG** [reply]
Makes sense @whizzer929, the question was certainly not to judge, more to understand the business logic of how to decide to pull a CIDR from the beginning or end of the range to see if that can be automated. From what I'm hearing here, this fairly manual and requires knowledge of the target application which IPAM wouldn't have context of that, per-se.

**Reply by DCMattyG** [reply]
Hey @whizzer929, someone else brought up a very similar (perhaps the same?) topic in #50.

Take a look [here](https://github.com/Azure/ipam/discussions/50#discussioncomment-4751185) and see if this would address the above for you as well.

Thanks!
